<launch>

  <arg name="twoWayDrive" default="true"/><!--false-->
  <arg name="maxSpeed" default="2.0"/><!--2.0-->
  <arg name="maxAccel" default="5.0"/>
  <arg name="cameraOffsetZ" default="0"/>
  <arg name="autonomyMode" default="true"/>
  <arg name="autonomySpeed" default="1.0"/><!--5.0-->
  <arg name="joyToSpeedDelay" default="1.0"/><!--2.0-->
  <arg name="goalX" default="0"/>
  <arg name="goalY" default="0"/>
  <arg name="goalZ" default="0"/>
  <arg name="v_kp" default="5"/>
  <arg name="v_ki" default="0"/>
  <arg name="v_kd" default="0.5"/>

  <node pkg="local_planner" type="localPlanner" name="localPlanner" output="screen" required="true">
    <param name="pathFolder" type="string" value="$(find local_planner)/paths" />
    <param name="vehicleLength" type="double" value="0.2" />
    <param name="vehicleWidth" type="double" value="0.2" />
    <param name="twoWayDrive" value="$(arg twoWayDrive)" />
    <param name="laserVoxelSize" type="double" value="0.05" /><!--0.05-->
    <param name="terrainVoxelSize" type="double" value="0.1" />
    <param name="useTerrainAnalysis" type="bool" value="true" />
    <param name="checkObstacle" type="bool" value="true" />
    <param name="checkRotObstacle" type="bool" value="false" /><!--false-->
    <param name="adjacentRange" type="double" value="2.1" />
    <param name="obstacleHeightThre" type="double" value="0.15" /><!--上坡:0.30  0.20-->
    <param name="groundHeightThre" type="double" value="0" /><!--0.15-->
    <param name="costHeightThre" type="double" value="0.1" />
    <param name="costScore" type="double" value="0.02" />
    <param name="useCost" type="bool" value="false" />
    <param name="pointPerPathThre" type="int" value="5" /><!--2-->
    <param name="minRelZ" type="double" value="-2.0" />
    <param name="maxRelZ" type="double" value="0.35" />
    <param name="maxSpeed" value="$(arg maxSpeed)" />
    <param name="dirWeight" type="double" value="1.0" /><!--0.02-->
    <param name="rotWeight" type="double" value="0.3" /><!--0.02-->
    <param name="slopeWeight" type="double" value="100.0" /><!--0.02-->
    <param name="dirThre" type="double" value="90.0" /><!--xz:90.0 xy:180 不能太大，会出问题，90正好-->
    <param name="dirToVehicle" type="bool" value="false" /><!--一定是false-->
    <param name="pathScale" type="double" value="0.35" />
    <param name="minPathScale" type="double" value="0.35" />
    <param name="pathScaleStep" type="double" value="0.2" /><!--0.1-->
    <param name="pathScaleBySpeed" type="bool" value="true" />
    <param name="minPathRange" type="double" value="0.4" />
    <param name="pathRangeStep" type="double" value="0.1" />
    <param name="pathRangeBySpeed" type="bool" value="true" />
    <param name="pathCropByGoal" type="bool" value="true" />
    <param name="autonomyMode" value="$(arg autonomyMode)" />
    <param name="autonomySpeed" value="$(arg autonomySpeed)" />
    <param name="joyToSpeedDelay" value="$(arg joyToSpeedDelay)" />
    <param name="joyToCheckObstacleDelay" type="double" value="5.0" />
    <param name="goalClearRange" type="double" value="0.35" />
    <param name="goalX" type="double" value="$(arg goalX)" />
    <param name="goalY" type="double" value="$(arg goalY)" />
    <param name="vehicleRadio" type="double" value="0.35" />
  </node>

  <node pkg="local_planner" type="pathFollower" name="pathFollower" output="screen" required="true">
    <param name="pubSkipNum" type="int" value="1" /><!--1-->
    <param name="twoWayDrive" value="false" />
    <param name="lookAheadDis" type="double" value="0.35" /><!--xz:0.5 xy:0.5-->
    <param name="yawRateGain" type="double" value="5.5" /><!--7.5-->
    <param name="stopYawRateGain" type="double" value="2.0" /><!--7.5-->
    <param name="maxYawRate" type="double" value="360.0" /><!--360  120.0-->
    <param name="maxSpeed" value="$(arg maxSpeed)" />
    <param name="maxAccel" value="$(arg maxAccel)" /><!--5.0-->
    <param name="switchTimeThre" type="double" value="0.3" />
    <param name="dirDiffThre" type="double" value="0.2" /><!--0.3-->
    <param name="stopDisThre" type="double" value="0.02" /><!--xz:0.2  xy:0.1-->
    <param name="slowDwnDisThre" type="double" value="2.0" /><!--0.3-->
    <param name="useInclRateToSlow" type="bool" value="true" /><!--false-->
    <param name="inclRateThre" type="double" value="10.0" /><!--120-->
    <param name="slowRate1" type="double" value="0.05" /><!--0.25-->
    <param name="slowRate2" type="double" value="0.15" /><!--0.5-->
    <param name="slowTime1" type="double" value="3.5" /><!--2.0-->
    <param name="slowTime2" type="double" value="3.5" /><!--2.0-->
    <param name="useInclToStop" type="bool" value="false" />
    <param name="inclThre" type="double" value="45.0" />
    <param name="stopTime" type="double" value="5.0" />
    <param name="noRotAtStop" type="bool" value="false" />
    <param name="noRotAtGoal" type="bool" value="true" />
    <param name="autonomyMode" value="$(arg autonomyMode)" />
    <param name="autonomySpeed" value="$(arg autonomySpeed)" />
    <param name="joyToSpeedDelay" value="$(arg joyToSpeedDelay)" />
    <param name="limitByAcc" value="true" />
    <param name="adjustByPitch" value="true" />
    <param name="goalX" type="double" value="$(arg goalX)" />
    <param name="goalY" type="double" value="$(arg goalY)" />
    <param name="goalZ" type="double" value="$(arg goalZ)" />
    <param name="v_kp" type="double" value="$(arg v_kp)" />
    <param name="v_ki" type="double" value="$(arg v_ki)" />
    <param name="v_kd" type="double" value="$(arg v_kd)" />
  </node>

</launch>